h/api/models/test/document_test.py

```diff
@@ -434,6 +435,203 @@ def merge_data(self, request):
         return (master, duplicate)
 
 
+class TestUpdateDocumentMetadata(object):
+
+    def test_it_uses_the_target_uri_to_get_the_document(self,
+                                                        annotation,
+                                                        Document,
+                                                        session):
+        document_uri_dicts = [
+            {
+                'uri': 'http://example.com/example_1',
+                'claimant': 'http://example.com/claimant',
+                'type': 'type',
+                'content_type': None,
+            },
+            {
+                'uri': 'http://example.com/example_2',
+                'claimant': 'http://example.com/claimant',
+                'type': 'type',
+                'content_type': None,
+            },
+            {
+                'uri': 'http://example.com/example_3',
+                'claimant': 'http://example.com/claimant',
+                'type': 'type',
+                'content_type': None,
+            },
+        ]
+
+        document.update_document_metadata(session,
+                                          annotation,
+                                          [],
+                                          document_uri_dicts)
+
+        Document.find_or_create_by_uris.assert_called_once_with(
+            session,
+            annotation.target_uri,
+            [
+                'http://example.com/example_1',
+                'http://example.com/example_2',
+                'http://example.com/example_3',
+            ],
+            created=annotation.created,
+            updated=annotation.updated,
+        )
+
+    def test_if_there_are_multiple_documents_it_merges_them_into_one(
+            self,
+            annotation,
+            Document,
+            merge_documents,
+            session):
+        """If it finds more than one document it calls merge_documents()."""
+        Document.find_or_create_by_uris.return_value = mock.Mock(
```

Comment:
Not going to block on this at this point, but I really think this would have been a prime candidate for just creating some data in the database and checking the end state was correct, rather than mocking out classmethods on `Document`.